web/src/components/index/UserTasks.tsx

```diff
@@ -1,81 +1,50 @@
 import { useAuth } from "@/hooks/useAuth";
 import LoginFallback from "@/components/ui/LoginFallback";
-import Button from "../ui/Button";
-import { FormEvent, useEffect, useRef, useState } from "react";
-import styles from "./UserTasks.module.css";
-import { postProfile, getProfile } from "@/openapi";
+import { useEffect, useState } from "react";
+import { getProfile } from "@/openapi";
+import ChosenNameForm from "./ChosenNameForm";
 
 export default function UserTasks() {
   const { loading, isAuthed, userData } = useAuth();
   const [chosenName, setChosenName] = useState<string | undefined>(undefined);
-  const dialogRef = useRef<HTMLDialogElement>(null);
 
   useEffect(() => {
-    try {
-      getProfile().then((response) => {
+    const fetchProfile = async () => {
+      try {
+        const response = await getProfile();
+
         if (response.response.ok && response.data?.chosen_name) {
-          setChosenName(response.data?.chosen_name);
+          setChosenName(response.data.chosen_name);
         } else {
           setChosenName("");
         }
```

Comment:
The original implementation mixed `try/catch` with the `.then` syntax which would not have caught error correctly. The `.then` approach has it's own way of catching errors and using `try/catch` is intended to be used with the `await` approach. 

I prefer using the `await` approach as I think it looks cleaner, so I've gone with that here.